src/index.js

```diff
@@ -1,35 +1,163 @@
+  input.value = ' '; // empty the input
 };
 
-const lists = [task1, task2, task3];
+// click on task elemnt to delete
+tasksDiv.addEventListener('click', (e) => {
+  // remove from page
+  if (e.target.classList.contains('del')) {
+    // remove from local
+    // eslint-disable-next-line no-use-before-define
+    deleteTaskWith(e.target.parentElement.getAttribute('data-id'));
+    // remove from page
+    e.target.parentElement.remove();
+  }
+
+  // task element and update
+  if (e.target.classList.contains('checked')) {
+    // toggele completed for the task
+    // eslint-disable-next-line no-use-before-define
+    toggleStatusTask(e.target.parentElement.getAttribute('data-id')); // here we don't call parent because we are on it
+    // toggle done class
+    e.target.classList.toggle('done');
+  }
+});
+
+function addTaskToArray(taskText) {
+  const task = {
+    id: Date.now(), // make it quall to time to be different
+    title: taskText,
+    completed: false,
+    index: arrayOfTasks.length,
+  };
+  // push my tasks to array
+  arrayOfTasks.push(task);
+
+  // Add elemnt to my page
+  // eslint-disable-next-line no-use-before-define
+  addElementsToPageFrom(arrayOfTasks);
+
+  // add to local storage
+  // eslint-disable-next-line no-use-before-define
+  addDataToLocal(arrayOfTasks);
+}
+
+function addElementsToPageFrom(arrayOfTasks) {
+  // empty task div if has any data
+  tasksDiv.innerHTML = ' ';
+
+  // looping arryof tasks
+
+  arrayOfTasks.forEach((task) => {
+    // create main div
+    const div = document.createElement('div');
+    div.className = 'task';
+
+    // check if task id done
+    if (task.completed) {
+      div.className = 'task done';
+    }
+    div.setAttribute('data-id', task.id);
+    // div.appendChild(document.createTextNode(task.title));
 
-const displayTasks = (lists) => {
-  lists.forEach((list) => {
-    const html = ` <div class"task-div"> <input type="checkbox" name="task" value="task">
-       <label class="list-item" for="task${list.index}"> ${list.description}</label><br>
-      </div>`;
+    // create checkbox
+    const checkbox = document.createElement('input');
+    checkbox.setAttribute('type', 'checkbox');
+    checkbox.className = 'checked';
+    checkbox.appendChild(document.createTextNode(''));
+    div.appendChild(checkbox);
 
-    listContainer.insertAdjacentHTML('beforeend', html);
+    // create input description
+    const desc = document.createElement('input');
+    desc.setAttribute('value', task.title);
+    desc.className = 'desc';
+    div.appendChild(desc);
+
+    // create delete span
+    const span = document.createElement('span');
+    span.className = 'del';
+    span.appendChild(document.createTextNode('Delete'));
+    // append button to div
+    div.appendChild(span);
+
+    desc.addEventListener('change', (e) => {
+      task.title = e.target.value;
+      // eslint-disable-next-line no-use-before-define
```

Comment:
- [ ] Please don't disable linter rules. These are here to write clean and standard code.